File: paymentsense-coding-challenge-api/Paymentsense.Coding.Challenge.Contracts/Response/GetCountryDetailResponse.cs
using Paymentsense.Coding.Challenge.Contracts.Dtos;
using System.Collections.Generic;

namespace Paymentsense.Coding.Challenge.Contracts.Response
{
    public class GetCountryDetailResponse : IResponse
    {
        public string Name { get; set; }
        public string Flag { get; set; }
        public int Population { get; set; }
        public IList<string> TimeZones { get; set; }
        public IList<CurrencyDto> Currencies { get; set; }
        public IList<LanguageDto> Languages { get; set; }
        public string CapitalCity { get; set; }
        public IList<string> BorderingCountries { get; set; }
    }
}

File: paymentsense-coding-challenge-website/src/app/models/queries/get-country-detail-response.ts
import { ICurrencyModel } from './../currency.model';
import { ILanguageModel } from './../language.model';

export interface IGetCountryDetailResponse {
    name : string;
    flag : string;
    population : number;
    timeZones : string[]
    currencies : ICurrencyModel[]
    languages : ILanguageModel[]
    capitalCity : string
    borderingCountries : string[]
}